community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/csv/InputGroupsDeserializer.java

```diff
@@ -39,8 +39,9 @@
     private final Header.Factory headerFactory;
     private final Configuration config;
     private final IdType idType;
-    private InputEntityDeserializer<ENTITY> currentInput;
+    private InputIterator<ENTITY> currentInput = new InputIterator.Adapter<>();
     private long previousInputsCollectivePositions;
+    private volatile boolean currentInputOpen;
```

Comment:
Does this class experience concurrent access? If so, then it'd be nice with some comments about what concurrency is expected, and how thread-safety is ensured.